### PR TITLE
Introduce FuzzyTime as component

### PR DESCRIPTION
**Pull request by vpereira**
Introduce a view component to deal with displaying date/time

### CODE REVIEW

File: src/api/app/helpers/webui/maintenance_incident_helper.rb [hunk 1]
```diff
@@ -111,7 +111,7 @@ def outgoing_request_links(requests)
             tag.i(nil, class: "fas fa-flag pr-1 request-flag-#{request['state']}", title: "Release request in state '#{request['state']}'")
           end,
           # rubocop:disable Rails/OutputSafety
-          "Created #{fuzzy_time(request.created_at)}".html_safe
+          FuzzyTimeComponent.new(time: request.created_at).human_time_ago.html_safe
```
Comment (vpereira) [inline src/api/app/helpers/webui/maintenance_incident_helper.rb:114]: I think we can safety change it for `sanitize` as described in the documentation https://apidock.com/rails/v6.1.3.1/String/html_safe. Even though the field `created_at` is never under user control.

File: src/api/app/views/webui/package/_revision_diff_footer.html.haml [hunk 2]
```diff
@@ -2,7 +2,7 @@
   %p
     The previous request
     = link_to(request['id'], request_show_path(project, package, request['id']))
-    was declined #{fuzzy_time_string(request['when'])} by
+    was declined #{FuzzyTimeComponent.new(time: request['when']).human_time_ago} by
```
Comment (vpereira) [inline src/api/app/views/webui/package/_revision_diff_footer.html.haml:5]: here we require just a string, so no template is going to be used.

File: src/api/app/views/webui/shared/bs_requests/index.json.erb [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "data": [
     <% table_rows.each.with_index do |row, index| %>
       [
-        "<%= escape_javascript(fuzzy_time(row.created_at, false)) %>",
+        "<%= escape_javascript(FuzzyTimeComponent.new(time: row.created_at).human_time_ago) %>",
```
Comment (vpereira) [inline src/api/app/views/webui/shared/bs_requests/index.json.erb:9]: we just need a string, no template.

File: src/api/spec/components/fuzzy_time_component_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  context 'time is in the past' do
+    let(:time) { 2.days.ago }
+
+    it { expect(render_inline(described_class.new(time: time)).to_html).to have_text('ago') }
+  end
+
+  context 'time is in the present' do
+    let(:time) { 1.second.ago }
+
+    it { expect(render_inline(described_class.new(time: time)).to_html).to have_text('now') }
```
Comment (dmarcoux) [inline src/api/spec/components/fuzzy_time_component_spec.rb:13]:
```suggestion
    it { expect(render_inline(described_class.new(time: time))).to have_text('now') }
```

File: src/api/app/components/fuzzy_time_component.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+class FuzzyTimeComponent < ApplicationComponent
+  attr_reader :time
+
+  def initialize(time:)
+    super
+    @time = time.utc
+  end
+
+  def human_time_ago
```
Comment (dmarcoux) [inline src/api/app/components/fuzzy_time_component.rb:11]: This method could be renamed to `call` and be an [inline view component](https://viewcomponent.org/guide/templates.html#inline) instead. Having a template for a single `<span>` element isn't needed.

**Comment by dmarcoux** [comment]
Please rebase the PR on master to get rid of the merge commits.

File: src/api/app/components/fuzzy_time_component.rb (hunk 5 above)
Comment (dmarcoux) [reply src/api/app/components/fuzzy_time_component.rb:11]: This comment still stands, could you address this? Thank you :slightly_smiling_face:

**Comment by dmarcoux** [comment]
> Please rebase the PR on master to get rid of the merge commits and squash the commits which came from suggestions.

Done!

**Review by dmarcoux — APPROVED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**